packages/jsii-pacmak/test/generated-code/__snapshots__/target-java.test.js.snap

```diff
@@ -27877,5 +27877,600 @@ exports[`Generated code for "jsii-calc": <outDir>/java/src/main/resources/softwa
++                .append(((java.util.List<java.lang.Object>)(value)).get(__idx_ac66f0).getClass())
++                .append(", (Parameter 'value')").toString());
++            }
++        }
+         software.amazon.jsii.Kernel.set(this, "unionArrayProperty", java.util.Objects.requireNonNull(value, "unionArrayProperty is required"));
+     }
+ 
+     /**
+      */
+@@ -234,12 +251,30 @@
+     }
+ 
+     /**
+      */
+     @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Stable)
+-    public void setUnionMapProperty(final @org.jetbrains.annotations.NotNull java.util.Map<java.lang.String, java.lang.Object> value) {
++    public void setUnionMapProperty(final @org.jetbrains.annotations.NotNull java.util.Map<java.lang.String, java.lang.Object> value) throws IllegalArgumentException {
+ 
++        for (java.util.Map.Entry<String, java.lang.Object> __item_ac66f0: ((java.util.Map<java.lang.String, java.lang.Object>)(value)).entrySet()) {
++            if (
++                 !(__item_ac66f0.getValue() instanceof java.lang.String)
++                && !(__item_ac66f0.getValue() instanceof java.lang.Number)
++                && !(__item_ac66f0.getValue() instanceof software.amazon.jsii.tests.calculator.lib.Number)
++                && !(__item_ac66f0.getValue().getClass().equals(software.amazon.jsii.JsiiObject.class))
++            )
++             {
++                throw new IllegalArgumentException(
++                new java.lang.StringBuilder("Expected ")
++                .append("value").append(".get(\\"").append((__item_ac66f0.getKey())).append("\\")")
++                .append(" to be one of: ")
++                .append("java.lang.String, java.lang.Number, software.amazon.jsii.tests.calculator.lib.Number")
++                .append("; received ")
++                .append(__item_ac66f0.getValue().getClass())
++                .append(", (Parameter 'value')").toString());
++            }
++        }
+         software.amazon.jsii.Kernel.set(this, "unionMapProperty", java.util.Objects.requireNonNull(value, "unionMapProperty is required"));
+     }
+ 
+     /**
+      */
+@@ -249,35 +284,35 @@
+     }
+ 
+     /**
+      */
+     @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Stable)
+-    public void setUnionProperty(final @org.jetbrains.annotations.NotNull java.lang.String value) {
++    public void setUnionProperty(final @org.jetbrains.annotations.NotNull java.lang.String value) throws IllegalArgumentException {
+ 
+         software.amazon.jsii.Kernel.set(this, "unionProperty", java.util.Objects.requireNonNull(value, "unionProperty is required"));
+     }
+ 
+     /**
+      */
+     @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Stable)
+-    public void setUnionProperty(final @org.jetbrains.annotations.NotNull java.lang.Number value) {
++    public void setUnionProperty(final @org.jetbrains.annotations.NotNull java.lang.Number value) throws IllegalArgumentException {
```

Comment:
something is wrong here...this `throws` should not be here, but I'm not sure what's causing it.